code/modules/research/fabricators.dm

```diff
@@ -62,21 +62,23 @@
 	max_material_storage = (initial(max_material_storage)+(T * 187500))
 
 	T = 0
-	for(var/obj/item/weapon/stock_parts/micro_laser/Ma in component_parts)
-		T += Ma.rating
-	if(T >= 1)
-		T -= 1
+	for(var/obj/item/weapon/stock_parts/manipulator/Ma in component_parts)
+		T += Ma.rating //T is the ammount of stock parts grade. 1 tier 1 part is a  T=1,  2 tier 2 parts is a T=4, 1 tier 1 and 1 tier 3 part is a T=5
+	if(T >= 2) // this removes 1 part tier level. This is to account for the fact that machines like the protolathe have 1 micro manip in them by default which means 1 tier level
+		T -= 2
```

Comment:
the multiply by 0 was intended by whoever originally coded it. A T1 part is worth 1 T and since most machines have 2 T1 parts the machine with these T1 parts would be counted as having 1 T2 part and as such would receive the unearned bonuses from being upgraded